agent/metrics/generic_metrics_client.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
there are no unit tests for functions in this file

Reply:
Added